kano_updater/commands/download.py

```diff
@@ -55,10 +55,29 @@ def download(progress=None, gui=True):
         progress.start('downloading')
 
     elif status.state == UpdaterStatus.UPDATES_DOWNLOADED:
-        err_msg = N_("Updates have been downloaded already")
-        logger.error(err_msg)
-        progress.abort(_(err_msg))
-        return True
+        progress.split(
+            Phase(
+                'checking',
+                _("Checking for updates"),
+                10,
+                is_main=True
+            ),
+            Phase(
+                'downloading',
+                _("Downloading updates"),
+                90,
+                is_main=True
+            )
+        )
+        progress.start('checking')
+        check_for_updates(progress=progress)
+        if status.state == UpdaterStatus.NO_UPDATES:
+            err_msg = N_("Latest updates have been downloaded already")
+            logger.info(err_msg)
+            progress.abort(_(err_msg))
+            return True
+
+        progress.start('downloading')
```

Comment:
It looks like this section is just the same as the `if status.state == UpdaterStatus.NO_UPDATES` case above with the exception of what it does if after checking there are no updates, would it be clearer to merge these two cases to

```python
if status.state in [UpdaterStatus.NO_UPDATES, UpdaterStatus.UPDATES_DOWNLOADED]:

    ...

    pre_check_state = status.state
    check_for_updates(progress=progress)

    if status.state == UpdaterStatus.NO_UPDATES:
        if pre_check_state == UpdaterStatus.NO_UPDATES:
            # No updates case
            return False
        if pre_check_state == UpdaterStatus.UPDATES_DOWNLOADED:
            # Updates downloaded case
            return True

    ...

elif status.state == UpdaterStatus.DOWNLOADING_UPDATES:
    ...
```

Maybe it wouldn't, but just a thought.

Reply:
I think it was initially easier for me to write it like that and clearer to anyone looking at it for the first time. I like the compressed format more now, let me have a go

Reply:
Fixed and tested